[ENH] provide proper mathematical description in docstring `ExpandingCutoffSplitter`

VascoSch92 opened the pull request:
#### Reference Issues/PRs
Partially solve #7196 


#### What does this implement/fix? Explain your changes.
Provide proper mathematical description in docstring

I should double-check if the mathematical description is correct. But I wanted to know if is that that you had in mind.


#### Any other comments?
I notice that the style of parameters section was different from the one used in `SlidingWindowSplitter`. So I fix it such that the style is equivalent in all the classes.

VascoSch92 commented:
I had also to change a little the given example. Otherwise, the documentation was not rendering correctly.

fkiraly reviewed (changes requested):
Excellent, great! Obviously you are en expert when it comes to math.

Some minor formatting things, requesting changes:

* in the parameter style we use (numpydoc I believe), there should be a space before the colon
* for code to be rendered in code font, you need to use double backticks (not single).
* if I understand you correctly, the splitter is purely `iloc` based and ignores `loc` index? If so, it is perhaps worth saying somewhere.

You can also check how the page renders by going on "details" next to the doc CI job.

VascoSch92 commented:
> in the parameter style we use (numpydoc I believe), there should be a space before the colon

Ok I will give a look

> for code to be rendered in code font, you need to use double backticks (not single).

Yes you are right :-). I forgot.

> if I understand you correctly, the splitter is purely iloc based and ignores loc index? If so, it is perhaps worth saying somewhere.

Good point. I will do.


> You can also check how the page renders by going on "details" next to the doc CI job.

I was creating the doc locally to check...

VascoSch92 commented:
- [x] in the parameter style we use (numpydoc I believe), there should be a space before the colon
- [x] for code to be rendered in code font, you need to use double backticks (not single).
- [ ] if I understand you correctly, the splitter is purely iloc based and ignores loc index? If so, it is perhaps worth saying somewhere.

For this last point: the old documentation of the class contained the following:

```
The valid types of y-index and cutoff pairings are datelike-datelike, datelike-int,
and int-int. When a datelike index is combined with an int cutoff, the cutoff
functions as an iloc indexer. When an int index is paired with a positive int
cutoff, the cutoff serves as a loc indexer. If the int cutoff is negative, it
functions as an iloc indexer.
```

Do you think it is enough or you would like something other?

fkiraly commented:
> Do you think it is enough or you would like something other?

Yes, this is enough, but then your math description seems incomplete.

fkiraly commented on sktime/split/expandingcutoff.py:
```diff
@@ -36,53 +36,85 @@ def _validate_cutoff(cutoff):
+    including all prior data up to the new cutoff.
+
+    .. math::
+        \text{TrainingFolds} = \{ [j, \ldots, cutoff + j]\}_{j=0, \ldots, N + 1}
```
this starts at j, is that correct? Should it not be the first time point?

Further, it might not be too clear that you are using indices. That may need some explanation, or, alternatively, introduction of a time series such as $s_1, \dots, s_T$ to make the value/index relation clear.

fkiraly reviewed (changes requested):
Great! See above though, I think there might be a mistake since your notation implies that the start also shifts (which I think it does not)

VascoSch92 commented:
> this starts at j, is that correct? Should it not be the first time point?
Further, it might not be too clear that you are using indices. That may need some explanation, or, alternatively, introduction ...

Yes you are right. It is wrong and I should say somewhere that the splitters return the indices for the splits ;-) 


I corrected the `TrainSplits` and `TestSplits` and I added at the beginning that the class returns the indices for the test and train splits. Let me know if it is enough or I should cite that somewhere else. 

Note that `TrainSplits` and `TestSplits` describe the indices of the train and test splits. Or would you like that they describe the time-series points? (i.e., I will add `s_` everywhere ;-) )

fkiraly commented:
> Or would you like that they describe the time-series points?

I think either is fine if it is clear from the setup (definitions) what the reference point is.

VascoSch92 commented:
> > Or would you like that they describe the time-series points?
> 
> I think either is fine if it is clear from the setup (definitions) what the reference point is.
> 
> Because `loc` based indexing can occur, I suppose it might be better to include the values plus indices in the setup (e.g., s i etc), but I am not certain.

The class is not always giving back just the indices?

Let me know which one you like so I can change ;-)

VascoSch92 commented:
@fkiraly ?
